beacon-chain/rpc/eth/light-client/handlers_test.go

```diff
@@ -1071,112 +1321,280 @@ func TestLightClientHandler_GetLightClientUpdatesByRange_TooBigCountAltair(t *te
+	st, err := util.NewBeaconStateAltair()
 	require.NoError(t, err)
+	headSlot := slot.Add(65)
+	err = st.SetSlot(headSlot)
+	require.NoError(t, err)
+
+	db := setupDB(t)
+
+	updates := make([]*ethpbv2.LightClientUpdate, 3)
+
+	updatePeriod := slot.Div(uint64(config.EpochsPerSyncCommitteePeriod)).Div(uint64(config.SlotsPerEpoch))
+
+	for i := 0; i < 3; i++ {
+		if i == 1 { // skip this update
+			updatePeriod++
+			continue
+		}
+		newSlot := slot.Add(uint64(i * 32))
+
+		update := &ethpbv2.LightClientUpdate{
+			AttestedHeader: &ethpbv2.LightClientHeaderContainer{
+				Header: &ethpbv2.LightClientHeaderContainer_HeaderAltair{
+					HeaderAltair: &ethpbv2.LightClientHeader{
+						Beacon: &ethpbv1.BeaconBlockHeader{
+							Slot:          newSlot,
+							ProposerIndex: 1,
+							ParentRoot:    []byte{1, 1, 1},
+							StateRoot:     []byte{1, 1, 1},
+							BodyRoot:      []byte{1, 1, 1},
+						},
+					},
+				},
+			},
+			NextSyncCommittee: &ethpbv2.SyncCommittee{
+				Pubkeys:         nil,
+				AggregatePubkey: nil,
+			},
+			NextSyncCommitteeBranch: nil,
+			FinalizedHeader:         nil,
+			FinalityBranch:          nil,
+			SyncAggregate: &ethpbv1.SyncAggregate{
+				SyncCommitteeBits:      []byte{1, 1, 1},
+				SyncCommitteeSignature: []byte{1, 1, 1},
+			},
+			SignatureSlot: 7,
+		}
+
+		updates[i] = update
+
+		err = db.SaveLightClientUpdate(ctx, uint64(updatePeriod), &ethpbv2.LightClientUpdateWithVersion{
+			Version: version.Altair,
+			Data:    update,
+		})
+		require.NoError(t, err)
+
+		updatePeriod++
+	}
 
-	h, err := signedBlock.Header()
-	require.NoError(t, err)
+	mockChainService := &mock.ChainService{State: st}
+	s := &Server{
+		HeadFetcher: mockChainService,
+		BeaconDB:    db,
+	}
+	startPeriod := 0
+	url := fmt.Sprintf("http://foo.com/?count=10&start_period=%d", startPeriod)
+	request := httptest.NewRequest("GET", url, nil)
+	writer := httptest.NewRecorder()
+	writer.Body = &bytes.Buffer{}
 
-	err = st.SetLatestBlockHeader(h.Header)
-	require.NoError(t, err)
-	stateRoot, err := st.HashTreeRoot(ctx)
-	require.NoError(t, err)
+	s.GetLightClientUpdatesByRange(writer, request)
 
-	// get a new signed block so the root is updated with the new state root
-	block.Block.StateRoot = stateRoot[:]
-	signedBlock, err = blocks.NewSignedBeaconBlock(block)
+	require.Equal(t, http.StatusOK, writer.Code)
+	var resp structs.LightClientUpdatesByRangeResponse
+	err = json.Unmarshal(writer.Body.Bytes(), &resp.Updates)
 	require.NoError(t, err)
+	require.Equal(t, 1, len(resp.Updates))
+	for i, update := range updates {
+		if i < 1 {
+			require.Equal(t, "altair", resp.Updates[i].Version)
+			updateJson, err := structs.LightClientUpdateFromConsensus(update)
+			require.NoError(t, err)
+			require.DeepEqual(t, updateJson, resp.Updates[i].Data)
+		}
+	}
+}
 
-	root, err := block.Block.HashTreeRoot()
-	require.NoError(t, err)
+func TestLightClientHandler_GetLightClientUpdatesByRangeMissingUpdates2(t *testing.T) {
```

Comment:
Maybe merge this test with the previous one and add a `t.Run` with a description?